### PR TITLE
[BUG] fix 4149, add sessionVariable to choose broadcastjoin first when cardinality cannot be estimated

### LINKED ISSUE #4149: [BUG] choose broadcastjoin first when cardinality cannot be estimated
sql：
`select * from (select k1 from jointest group by k1)t2, jointest t1 where t1.k1 = t2.k1`

reason:
In version 0.11, the SQL choose broadcastjoin and returns it in 9s. However, in version 0.12, the SQL choose shuffle join and runs 32S.
When the cardinality cannot be estimated, it is better to choose broadcast join first. When there is a performance bottleneck, users can manually add hint to shuffle point.

### PR DESCRIPTION
**Pull request by wutiangan**
fix #4149

add sessionVariable "prefer_broadcast_join" to choose broadcastjoin first when cardinality cannot be estimated
the sessionVariable can set local or global. the default value is true, prefer broadcast join first where the two cost is equal.

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]
+1

File: docs/zh-CN/administrator-guide/variables.md [hunk 1]
```diff
@@ -338,3 +338,6 @@ SET forward_to_master = concat('tr', 'u', 'e');
 
     是否将 bitmap 和 hll 类型的 count distinct 查询重写为 bitmap_union_count 和 hll_union_agg 。
 
+* `prefer_broadcast_join`
```
Comment (morningman) [inline docs/zh-CN/administrator-guide/variables.md:341]: How about name it to "prefer_join_method", and value is "broadcast" or "shuffle"

**Review by morningman — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/planner/DistributedPlanner.java [hunk 2]
```diff
@@ -286,7 +286,8 @@ private PlanFragment createScanFragment(PlanNode node) {
      * Therefore, we add a SessionVariable to help users choose a better join implementation.
      */
     private boolean isBroadcastCostSmaller(long broadcastCost, long partitionCost)  {
-        if (ConnectContext.get().getSessionVariable().isPreferBroadcastJoin()) {
+        String joinMethod = ConnectContext.get().getSessionVariable().getPreferJoinMethod();
+        if (joinMethod.equals("broadcast")) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/org/apache/doris/planner/DistributedPlanner.java:290]: `equalsIgnoreCase` should more better.

**Review by kangkaisen — APPROVED** [review]
+1

**Pull request merged**